lib/AST/DiagnosticEngine.cpp

```diff
@@ -140,6 +143,39 @@ struct EducationalNotes {
 static constexpr EducationalNotes<LocalDiagID::NumDiags> _EducationalNotes = EducationalNotes<LocalDiagID::NumDiags>();
 static constexpr auto educationalNotes = _EducationalNotes.value;
 
+class LocalizationInput : public llvm::yaml::Input {
```

Comment:
@xedin I created the `LocalizationInput` class here in the `.cpp` file. Do you think this has to move to the `.h` file? and if so... how will we be able to write the implementation of the `>>` operator in the `.cpp` file?

Because I think if we moved the class declaration to the `.h` file. We will need to make the `>>` operator a non-member operator. Love to hear your thoughts. :)

Reply:
Since it's only used in this file it's okay to declare it here.